src/main/java/de/tum/in/www1/artemis/service/exam/StudentExamService.java

```diff
@@ -486,9 +477,53 @@ public void setUpExerciseParticipationsAndSubmissions(StudentExam studentExam, L
         }
     }
 
+    /**
+     * Starts all the exercises of all the student exams of an exam
+     *
+     * @param examId exam to which the student exams belong
+     * @return number of generated Participations
+     */
+    public int startExercises(Long examId) {
+        var exam = examRepository.findWithStudentExamsExercisesById(examId).orElseThrow(() -> new EntityNotFoundException("Exam", examId));
+        var studentExams = exam.getStudentExams();
+        List<StudentParticipation> generatedParticipations = Collections.synchronizedList(new ArrayList<>());
```

Comment:
As far as I can see, the list is only used for counting. Could you also just increment an AtomicInt?

Reply:
This method was not implemented by me  [(see StudentExamSerive:420)](https://github.com/ls1intum/Artemis/blob/develop/src/main/java/de/tum/in/www1/artemis/service/exam/StudentExamService.java). 
The change may occurred, as I implemented a new Method  `setUpExerciseParticipationsAndSubmissionsWithInitializationDate`, which reuses the implemented functionality of the old `setUpExerciseParticipationsAndSubmissions`. The new one `setUpExerciseParticipationsAndSubmissions` (line 519) calls the `[...]WithInitializationDate` without passing an initialization Date.

Reply:
Since you're already refactoring, I thought it might be a good idea to adapt the copied code as well